src/common/pjrt_implementation/executable_image.cc

```diff
@@ -0,0 +1,128 @@
+    // This should be updated once iree supports replicas.
+    args->num_replicas = 1;
+    return nullptr;
+  };
+  api->PJRT_Executable_Serialize =
+      +[](PJRT_Executable_Serialize_Args *args) -> PJRT_Error * {
+    DLOG_F(LOG_DEBUG, "ExecutableImage::PJRT_Executable_Serialize");
+    return ErrorInstance::MakeError(tt_pjrt_status::kUnimplemented);
+  };
+  api->PJRT_Executable_DeserializeAndLoad =
+      +[](PJRT_Executable_DeserializeAndLoad_Args *args) -> PJRT_Error * {
+    DLOG_F(LOG_DEBUG,
+           "ExecutableImage::PJRT_Executable_DeserializeAndLoad_Args");
+    return ErrorInstance::MakeError(tt_pjrt_status::kUnimplemented);
+  };
+  api->PJRT_Executable_Serialize =
+      +[](PJRT_Executable_Serialize_Args *args) -> PJRT_Error * {
+    DLOG_F(LOG_DEBUG, "ExecutableImage::PJRT_Executable_Serialize_Args");
+    return ErrorInstance::MakeError(tt_pjrt_status::kUnimplemented);
+  };
+  api->PJRT_Executable_OptimizedProgram =
+      +[](PJRT_Executable_OptimizedProgram_Args *args) -> PJRT_Error * {
+    DLOG_F(LOG_DEBUG, "ExecutableImage::PJRT_Executable_OptimizedProgram");
+    ExecutableImage *executable = ExecutableImage::Unwrap(args->executable);
+    PJRT_Program *program = args->program;
+    program->format = kMlirFormat.data();
+    program->format_size = kMlirFormat.size();
+    size_t code_size = executable->code.size();
+    if (program->code == nullptr) {
+      program->code_size = code_size;
+    } else {
+      if (program->code_size < code_size) {
+        return ErrorInstance::MakeError(tt_pjrt_status::kInvalidArgument);
+      }
+      std::memcpy(program->code, executable->code.c_str(),
+                  executable->code.size());
+    }
+    return nullptr;
+  };
+  api->PJRT_Executable_GetCostAnalysis =
+      +[](PJRT_Executable_GetCostAnalysis_Args *args) -> PJRT_Error * {
+    DLOG_F(LOG_DEBUG, "ExecutableImage::PJRT_Executable_GetCostAnalysis_Args");
+    return ErrorInstance::MakeError(tt_pjrt_status::kUnimplemented);
+  };
+  api->PJRT_Executable_OutputElementTypes =
+      +[](PJRT_Executable_OutputElementTypes_Args *args) -> PJRT_Error * {
+    DLOG_F(LOG_DEBUG,
+           "ExecutableImage::PJRT_Executable_OutputElementTypes_Args");
+    return ErrorInstance::MakeError(tt_pjrt_status::kUnimplemented);
+  };
+  api->PJRT_Executable_OutputDimensions =
+      +[](PJRT_Executable_OutputDimensions_Args *args) -> PJRT_Error * {
+    DLOG_F(LOG_DEBUG, "ExecutableImage::PJRT_Executable_OutputDimensions_Args");
+    return ErrorInstance::MakeError(tt_pjrt_status::kUnimplemented);
+  };
+  api->PJRT_Executable_OutputMemoryKinds =
+      +[](PJRT_Executable_OutputMemoryKinds_Args *args) -> PJRT_Error * {
+    DLOG_F(LOG_DEBUG, "ExecutableImage::PJRT_Executable_OutputMemoryKinds");
+    return ErrorInstance::MakeError(tt_pjrt_status::kUnimplemented);
+  };
+}
+
+//===----------------------------------------------------------------------===//
+// Top-level API binding.
```

Comment:
Seems like a stray comment left from the original `api_impl.cc`, probably meant to be in `api_bindings.h`